sqlmesh/core/engine_adapter/clickhouse.py

```diff
@@ -296,6 +293,70 @@ def _build_partitioned_by_exp(
             this=exp.Schema(expressions=partitioned_by),
         )
 
+    def inject_query_setting(
+        self,
+        query: Query,
+        setting_name: t.Optional[str] = "join_use_nulls",
+        setting_value: t.Optional[str] = "1",
```

Comment:
I don't think it's good practice to change the prototypes of methods, even if it's default values. Let's keep the definition consistent and do something like `setting_name = setting_name or "join_use_nulls` at the beginning of this method.

Reply:
Thank you - that felt wrong, but I couldn't quite see the alternative